ocaml/database/db_cache_types.ml

```diff
@@ -126,6 +147,17 @@ functor
 module Row = struct
   include Make (Schema.Value)
 
+  let add gen key v =
+    add gen key
+    @@
+    match v with
+    | Schema.Value.String x ->
+        Schema.Value.String (unique x)
+    | Schema.Value.Pairs ps ->
+        Schema.Value.Pairs (List.map (fun (x, y) -> (unique x, unique y)) ps)
```

Comment:
You could change `add_to_map` to do the unique call, so we don't have to walk the map twice.